api/v1alpha4/openstackmachinetemplate_webhook.go

```diff
@@ -51,9 +53,17 @@ func (r *OpenStackMachineTemplate) ValidateCreate() error {
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type.
 func (r *OpenStackMachineTemplate) ValidateUpdate(old runtime.Object) error {
+	var allErrs field.ErrorList
 	oldOpenStackMachineTemplate := old.(*OpenStackMachineTemplate)
-	if !reflect.DeepEqual(r.Spec, oldOpenStackMachineTemplate.Spec) {
-		return errors.New("openstackMachineTemplateSpec is immutable")
```

Comment:
um...  looks like we did this kind of check before here? @sbueringer from your original issue
seems we still allow such changes before this PR? or we return wrong error ?

Reply:
I actually wasn't aware that we already had this in place :)

But I think the new version is nicer and probably in line with other providers @johannesfrey ?

Reply:
Yes I agree this is better :)

the question is whether we had problem before (by using existing code)? if so then with new PR, the problem might not be fixed?

Reply:
No it's completely fine imho. I'll test if it actually works. My issue was not triggered because I saw the problem in practice. I had a discussion the cluster-api Slack and then I thouht "well we also need this", and wasn't aware that we already had an implementation for it.

Reply:
The only change is that with this PR we don't compare the `OpenStackMachineTemplate.Spec` anymore but rather `OpenStackMachineTemplate.Spec.Template.Spec`. This is also done in [CAPZ](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1275/files#diff-5acce21862b7c3e7b1ca6b85ba34ae2a798a488972d16ea6275655beb1053fc0R58). Is this correct or should we rather block changes to `OpenStackMachineTemplate.Spec` like before?

Reply:
@johannesfrey Good point. I'll take a closer look next week and try to find out

Reply:
I think we are using deep copy anyway?

Reply:
I gave a try here;

```
root@jitest19:~/abc# kubectl edit openstackmachinetemplate
openstackmachinetemplate.infrastructure.cluster.x-k8s.io/capi-openstack-control-plane skipped
error: openstackmachinetemplates.infrastructure.cluster.x-k8s.io "capi-openstack-md-0" could not be patched: admission webhook "validation.openstackmachinetemplate.infrastructure.x-k8s.io" denied the request: openstackMachineTemplateSpec is immutable
You can run `kubectl replace -f /tmp/kubectl-edit-fh9fo.yaml` to try this update again.
root@jitest19:~/abc#
```

Reply:
@jichenjc what does deepcopy have to do with this PR? :)

Reply:
@sbueringer we are using deep copy of openstackmachinetemplate.Spec before ? 
so actually previous code works fine as I showed above... so this PR's commit msg is not accurate
maybe an improvement on the wording but not provide immutability enhancement as we already have it..

Reply:
I guess you mean deepequal?

Reply:
my bad .. yes, I thought deepequal but typed deepcopy ..